src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayObject = [];
+  const updateObject = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(updateObject, action.extraData);
+      arrayObject.push({ ...updateObject });
+    } else if (action.type === 'removeProperties') {
+      for (const keyRemove of action.keysToRemove) {
+        delete updateObject[keyRemove];
+      }
+      arrayObject.push({ ...updateObject });
+    } else if (action.type === 'clear') {
+      for (const key in updateObject) {
+        delete updateObject[key];
+      }
```

Comment:
avoid such nesting `else if` conditions
This code is hard to read and maintain

Reply:
use `switch case` for such predefined cases

Reply:
fixed